Source/NSPathControl.m

```diff
@@ -0,0 +1,396 @@
+/* Implementation of class NSPathControl
+   Copyright (C) 2020 Free Software Foundation, Inc.
+   
+   By: Gregory John Casamento
+   Date: Wed Apr 22 18:19:40 EDT 2020
+
+   This file is part of the GNUstep Library.
+   
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+   
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+   
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110 USA.
+*/
+
+#import "AppKit/NSPathControl.h"
+#import "AppKit/NSPathCell.h"
+#import "AppKit/NSGraphics.h"
+#import "AppKit/NSDragging.h"
+#import "AppKit/NSPasteboard.h"
+#import "AppKit/NSMenu.h"
+#import "AppKit/NSOpenPanel.h"
+
+@implementation NSPathControl
+
++ (void) initialize
+{
+  if (self == [NSPathControl class])
+    {
+      [self setVersion: 1.0];
+      [self setCellClass: [NSPathCell class]];
```

Comment:
With the current implementation of NSControl you will need to override the setCellClass: method locally. You may consider this a bug in GNUstep gui. Or see it as a convention that you need to follow.
I am really not sure how Cocoa implements that. You would need to write a subclass of NSControl and test the behaviour.